### PR TITLE
[CP-858] Fix test security

### PR DESCRIPTION
**Pull request by pawel-mudita**
Add security EP API to QAMuditaHarness repo.
Refactor security EP test to make it work.

### CODE REVIEW

File: test/pytest/service-desktop/test_security.py [hunk 1]
```diff
@@ -1,33 +1,73 @@
...
+        self.harness = harness
+
+    def assert_phone_unlocked(self):
+        assert GetPhoneLockStatus().run(self.harness).response.status == status["NoContent"]
+
+    def assert_phone_locked(self):
+        try:
+            GetPhoneLockStatus().run(self.harness).response.status
+        except TransactionError as error:
+            assert error.status == status["Forbidden"]
```
Comment (pholat) [inline test/pytest/service-desktop/test_security.py:25]: consider logging the error explaining why does it happen

File: test/pytest/service-desktop/test_security.py [hunk 2]
```diff
@@ -1,33 +1,73 @@
...
+class SecurityTester:
+    def __init__(self, harness):
+        self.harness = harness
+
+    def assert_phone_unlocked(self):
+        assert GetPhoneLockStatus().run(self.harness).response.status == status["NoContent"]
+
+    def assert_phone_locked(self):
+        try:
+            GetPhoneLockStatus().run(self.harness).response.status
```
Comment (pholat) [inline test/pytest/service-desktop/test_security.py:23]: there are general fixtures for QAMuditaOS pytest- please change the usage there too.

File: test/pytest/service-desktop/test_security.py [hunk 3]
```diff
@@ -1,33 +1,73 @@
...
 import time
 
 from harness.interface.defs import status
-
+from harness.api.developermode import PhoneModeLock
+from harness.api.security import GetPhoneLockStatus, GetPhoneLockTime, SetPhoneLockOff
+from harness.request import TransactionError
+
+
+class SecurityTester:
```
Comment (pholat) [inline test/pytest/service-desktop/test_security.py:14]: please either:
1. move asserts to test
2. make this class a test class

File: test/pytest/service-desktop/test_security.py [hunk 4]
```diff
@@ -1,33 +1,73 @@
...
+            GetPhoneLockTime().run(self.harness)
+        except TransactionError as error:
+            assert error.status == status["UnprocessableEntity"]
+        else:
+            logging.error("Phone is time locked unexpectedly!")
+            assert False
+
+    def unlock_phone(self, passcode: list = None):
+        assert SetPhoneLockOff(passcode).run(self.harness).response.status == status["NoContent"]
+        time.sleep(0.5)
```
Comment (pholat) [inline test/pytest/service-desktop/test_security.py:50]: why is this sleep required?

Comment (pawel-mudita) [reply test/pytest/service-desktop/test_security.py:50]: On a physical device unlocking the phone takes (a short) time.
Without any sleep in this place, this operation could not be finished in the OS, before check after it.
0.1 or even less would be enough probably, but a bit more is reserved just in case.
Probably it could be combined with await and _assert_phone_locked_, but not sure if it's needed.
This solution is simple and reliable enough. The state of the lock is always checked after this operation, and if sleeping for 0.5 is not enough, this means that something is bad.
tl;dr
ihmo its reliable enough for this test

File: test/pytest/service-desktop/test_security.py (hunk 2 above)
Comment (pawel-mudita) [reply test/pytest/service-desktop/test_security.py:23]: Do you mean fixtures from conftest.py, like _@pytest.mark.usefixtures("phone_locked")_?
They are using developer mode EP API, not security EP API -> no changes are needed

File: test/pytest/service-desktop/test_security.py (hunk 1 above)
Comment (pawel-mudita) [reply test/pytest/service-desktop/test_security.py:25]: Actually, this is the expected status.
Just need to catch it here because some external class raises an exception on any status but _OK_

File: test/pytest/service-desktop/test_security.py (hunk 4 above)
Comment (pholat) [reply test/pytest/service-desktop/test_security.py:50]: we have a lot of tests like that and in my opinion its a bad practice.
1. we are waiting for something which is a side-effect but user doesn't know what it is - we should wait for event explicitly
2. it can be slightly delayed and it will be issue, when it should be. Or it would be frequently used, but will happen much faster - but code will be super slow as there would be lots of sleep code in it.

changing this to something we could trust should be fairly easy. Unfortunatelly this is quite common pattern in our tests. I could agree to ie. helping you out changing it to something better (few hour top)?

File: test/pytest/service-desktop/test_security.py (hunk 3 above)
Comment (pawel-mudita) [reply test/pytest/service-desktop/test_security.py:14]: Asserts moved to the test

File: test/pytest/service-desktop/test_security.py (hunk 4 above)
Comment (pawel-mudita) [reply test/pytest/service-desktop/test_security.py]: Will be changed in the separate task: https://appnroll.atlassian.net/browse/CP-953

File: test/pytest/service-desktop/test_security.py (hunk 1 above)
Comment (pholat) [reply test/pytest/service-desktop/test_security.py]: but we assert it here. so it's an expected failure? :)

Comment (pholat) [reply test/pytest/service-desktop/test_security.py]: please double check and close this comment :manbow:

**Review by pholat — APPROVED** [review]

File: test/pytest/service-desktop/test_security.py (hunk 1 above)
Comment (pawel-mudita) [reply test/pytest/service-desktop/test_security.py]: Yes.
It is a failure just from the perspective of the external class, which raises an exception if the status is different than `OK`.
Here we expect the status to be `Forbidden`, since the phone should be locked when we use this method.
See updated code, now assertion is moved to test. It should be clear now.

**Review by wojtekrzepecki — APPROVED** [review]
